tuf/repository_lib.py

```diff
@@ -1363,10 +1372,9 @@ def _generate_targets_fileinfo(target_files, targets_directory,
 
 
 
-
 def generate_snapshot_metadata(metadata_directory, version, expiration_date,
-    targets_filename, storage_backend, consistent_snapshot=False,
-    repository_name='default'):
+    storage_backend, consistent_snapshot=False, repository_name='default',
+    use_length=True, use_hashes=True):
```

Comment:
I'm still confused about what is public API and what is not... but is changing the signature like this appropriate here? I guess this is an "internal library" so it's fine?

There's other 'API' changes in the PR of course but this is the one that changes a signature in a potentially subtle way

Reply:
Well, I don't see another way to achieve the optional length and hashes without passing those additional arguments as 
`_generate_and_write_metadata` is not a member function of any class.

Reply:
So my concern was not with the added arguments but the removed one (which in some cases could create a situation where a caller keeps using old signature and it happens to get accepted by accident). But since I think this function is not API (_right?_) that's fine: you've fixed all of the call sites in tuf so it's all good.

Reply:
I don't think this function is an API.
The only other place where is called or referenced is in the `_generate_and_write_metadata` function which is an internal function.
Are we right that this is not a pr @lukpueh @joshuagl ?

Reply:
AFAIK python-TUF does not have a clearly defined API. The closest thing to an API definition is the TUF tutorial, which suggests that a user should interact with the TUF repository mainly through repository_tool.